cmd/gardenlet/app/gardenlet.go

```diff
@@ -353,6 +356,12 @@ func NewGardenlet(ctx context.Context, cfg *config.GardenletConfiguration) (*Gar
 		return nil, errors.New("unable to extract Gardener`s cluster identity from cluster-identity ConfigMap")
 	}
 
+	// create the certificate manager to schedule certificate rotations
+	var certificateManager *certificate.Manager
+	if cfg.GardenClientConnection != nil && cfg.GardenClientConnection.KubeconfigSecret != nil && cfg.SeedConfig != nil {
```

Comment:
Same here, `cfg.GardenClientConnection` can never be nil.
Also, can you remove the check for the seed config to make it the same like in line 274? I think the indicator that the gardenlet should use the bootstrapping+rotation mechanisms should only be if `KubeconfigSecret` is set and not the `SeedConfig`.
```suggestion
	if cfg.GardenClientConnection.KubeconfigSecret != nil {
```